### DISCUSSION TITLE
Template sensor for net_price_ct_per_kwh

### DISCUSSION THREAD

**Discussion by aichingerk** [mampfes/ha_epex_spot#70] [Q&A]
Hi,
I have a template sensor for the best time between 21:00 - 06:00  but also need a sensor for the price between the timeframe.

`# um zu wissen wenn in der Nacht 1,5h der beste Preis ist
trigger:
  - platform: time
    at: "15:00:00" # nach dem holen der neuen Daten
action:
  - service: epex_spot.get_lowest_price_interval
    data:
      earliest_start: "21:00:00"
      latest_end: "06:00:00"
      duration:
        hours: 1
        minutes: 30
    response_variable: resp
sensor:
  - name: epex_low_intervall_night_1_5h
    unique_id: epex_low_intervall_night_1_5h
    device_class: timestamp
    state: "{{ resp.start is defined and resp.start }}"`

And now I want to make a template sensor for the same time, but with the attribute of price "net_price_ct_per_kwh:" that I can use it in automation. Here I don't know how to set the device_class and state of the sensor?

sensor:
  - name: epex_low_intervall_night_1_5h_price
    unique_id: epex_low_intervall_night_1_5h_price
    device_class: ?
    state: ?

Can anyone give me a tip !

Thx aichingerk

**Comment by mampfes** [comment]
(not tested):
```
    sensor:
      - name: net price
        state: "{{ resp.net_price_ct_per_kwhis defined and resp.net_price_ct_per_kwh}}"
```

The available device classes are described here: https://developers.home-assistant.io/docs/core/entity/sensor/#available-device-classes

**Reply by aichingerk** [reply]
It is working, thx
